azure-kusto-ingest/src/resource_manager/resource_uri.rs

```diff
@@ -0,0 +1,193 @@
+use azure_core::ClientOptions;
+use azure_storage::StorageCredentials;
+use azure_storage_blobs::prelude::{ClientBuilder, ContainerClient};
+use azure_storage_queues::{QueueClient, QueueServiceClientBuilder};
+use url::Url;
+
+use anyhow::Result;
+
+/// Parsing logic of resource URIs as returned by the Kusto management endpoint
+#[derive(Debug, Clone)]
+pub struct ResourceUri {
+    service_uri: String,
+    object_name: String,
+    sas_token: StorageCredentials,
+}
+
+impl ResourceUri {
+    pub fn service_uri(&self) -> &str {
```

Comment:
comment: Is it worth having these accessor methods, when you could just make the fields `pub` (or `pub(crate)`)?  You could argue that they hide the internal implementation, but there's not much to hide, given that all the parsing is done on creation.